packages/docgen/src/compile.js

```diff
@@ -0,0 +1,47 @@
+const ts = require( 'typescript' );
+const { readFileSync } = require( 'fs' );
+
+const { getExportStatements } = require( './get-export-statements' );
+
+/**
+ * Function that takes file path and returns compiled result
+ *
+ * @param {string} filePath the path to the file to be compiled
+ */
+module.exports = function( filePath ) {
+	const options = {
+		allowJs: true,
+	};
+	const host = ts.createCompilerHost( options, true );
+	const raw = readFileSync( filePath ).toString();
+	const code = raw
+		// typescript interprets @wordpress in @example code as a JSDoc tag.
+		// So, it should be replaced for the time being.
+		// They're restored in get-jsdoc-from-statement.js
+		.replace( /@wordpress/g, '__WORDPRESS_IMPORT__' )
+		// When <caption>ES(5|Next)<\/caption> exists next to @example tag,
+		// typescript cannot parse code correctly.
+		// So, they're removed.
+		.replace( /@example\s+<caption>ES(5|Next)<\/caption>/g, '@example' );
```

Comment:
It'd probably be fine to address separately, but our usage of HTML after the `@example` tag feels weird. I wonder if it would help avoid this logic if we at least moved those `<caption>` tags onto the line immediately following `@example` ?

Reply:
Yes, it does. [Check this out](https://astexplorer.net/#/gist/959d69eaf758ea2281438e5a2b510bba/972255808da8964e869afea19e795e17ca1ce0cc) 

But I think we should add it to the 'later works' list. To fix it, we need to change 10 more files.

Reply:
> But I think we should add it to the 'later works' list. To fix it, we need to change another 10 more files.

What I find can work well in these situations is: Create a separate pull request which includes those smaller independent changes. At the time it's merged, the branch of this pull request can be rebased to account for those new revisions. For this case, I expect it means we wouldn't need this code anymore to handle this syntax.